File: GXpert/GXpert.Web/Modules/Syllabus/Course/CourseColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Syllabus.Columns;

[ColumnsScript("Syllabus.Course")]
[BasedOnRow(typeof(CourseRow), CheckNames = true)]
public class CourseColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Title { get; set; }
    public short SortOrder { get; set; }
    public string Thumbnail { get; set; }
    public string Description { get; set; }
    
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Syllabus/CourseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CourseRow } from "./CourseRow";

export interface CourseColumns {
    Id: Column<CourseRow>;
    Title: Column<CourseRow>;
    SortOrder: Column<CourseRow>;
    Thumbnail: Column<CourseRow>;
    Description: Column<CourseRow>;
}

export class CourseColumns extends ColumnsBase<CourseRow> {
    static readonly columnsKey = 'Syllabus.Course';
    static readonly Fields = fieldsProxy<CourseColumns>();
}